### DISCUSSION TITLE
Youtube app ads

### DISCUSSION THREAD

**Discussion by DistractedCanuck** [0xERR0R/blocky#178] [Q&A] [answered]
Hi, So I'm strongly considering this as an alternative to pi-hole for my home server.  I like the groups, as well as the built in support for DoT. (If I can get it to work).  That being said when testing out both pi-hole and blocky I haven't been able to stop adds on the YouTube app on my phone using Blocky.  They are by far the ads that bother me the most and I'd love to have them stopped.  I tried adding all the ad block lists from https://firebog.net/ but no luck.  Has anyone had any luck with this? Thanks!

**Comment by 0xERR0R** [comment]
Hi. Thanks for testing. The short answer is: you can't block youtube ads. The long answer: blocky uses (like pihole) [DNS Sinkhole](https://en.wikipedia.org/wiki/DNS_sinkhole) as blocking methodology. That means, some domains (like doublecklick.net) will be blocked on the DNS level and your client can't reach them. Youtube ads are loaded by the youtube client (e.g. youtube app) from youtube domain. If you block this domain, you can't use youtube anymore.

**Comment by DistractedCanuck** [comment]
Thanks for the reply! Makes sense.  I wonder why the ads on the youtube app were blocked when I used the pi-hole.

**Comment by 0xERR0R** [comment, marked as answer]
Are you sure, that it worked? According to this thread: https://discourse.pi-hole.net/t/how-do-i-block-ads-on-youtube/253/24 it is partially possible to block some domains, but is changes very fast. I know, some youtube client (e.g. youtube-dl) can "skip" ads, but I'm not sure, if it is possible on the DNS side. Client has more information (like whole url) and can implement some youtube specific logic

**Reply by DistractedCanuck** [reply]
Interesting.  It's seems like it's working again.  Maybe.  Might just be a fluk.  I'll have to run with it for a little while and see if any ads come up. I'll let you know.

**Reply by 0xERR0R** [reply]
Yes, please let me know if it works and which block list was used

**Reply by DistractedCanuck** [reply]
Yup as soon as I sent that I switched back to the pi-hole and got an ad on the first video.   Must have just been luck or I had already watched enough ads before.  

Great work on this btw.  I love how clean and simple blocky is. The DoT integration is also fantastic. Much easier than having to set up another DNS server between Blocky and external.  

Keep up on the great work!
